app/src/main/java/com/infomaniak/drive/data/api/ApiRoutes.kt

```diff
@@ -97,6 +98,45 @@ object ApiRoutes {
     fun trashURLV2(file: File) = "${driveURLV2(file.driveId)}/trash/${file.id}"
     private fun trashURL(file: File) = "${driveURL(file.driveId)}/trash/${file.id}"
 
+    /** File Url */
+    //region File url
+    fun getDownloadFileUrl(file: File): String = with(file) {
+        val downloadUrl = if (isPublicShared()) {
+            downloadPublicShareFile(driveId, publicShareUuid, id)
+        } else {
+            downloadFile(file)
+        }
+
+        return if (isOnlyOfficePreview()) "$downloadUrl?as=pdf" else downloadUrl
+    }
```

Comment:
What about doing extensions for these methods ?

For example:

```kotlin
fun File.getDownloadUrl(): String {
  val downloadUrl = if (isPublicShared()) {
      downloadPublicShareFile(driveId, publicShareUuid, id)
  } else {
      downloadFile(this)
  }

  return if (isOnlyOfficePreview()) "$downloadUrl?as=pdf" else downloadUrl
}
```

Reply:
If we put it as extension, we should put them in File.kt, that's a never ending question 😅